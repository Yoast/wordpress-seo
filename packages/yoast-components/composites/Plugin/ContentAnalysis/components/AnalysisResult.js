export { AnalysisResult as default } from "@yoast/analysis-report";
