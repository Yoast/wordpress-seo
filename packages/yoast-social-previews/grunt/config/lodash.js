// See https://github.com/lodash-archive/grunt-lodash
module.exports = {
	templates: {
		dest: "js/templates.js",
		options: {
			exports: [ "node" ],
			template: "<%= files.templates %>",
			flags: [ "--development" ],
			moduleId: "none",
		},
	},
};
