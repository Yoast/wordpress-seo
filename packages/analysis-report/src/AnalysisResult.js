import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { noop } from "lodash";

import { SvgIcon, IconButtonToggle, IconCTAEditButton, BetaBadge } from "@yoast/components";

const AnalysisResultBase = styled.li`
	// This is the height of the IconButtonToggle.
	min-height: 24px;
	padding: 0;
	display: flex;
	align-items: flex-start;
`;

const ScoreIcon = styled( SvgIcon )`
	margin: 3px 11px 0 0; // icon 13 + 11 right margin = 24 for the 8px grid.
`;

const AnalysisResultText = styled.p`
	margin: 0 16px 0 0;
	flex: 1 1 auto;
	color: ${ props => props.suppressedText ? "rgba(30,30,30,0.5)" : "inherit" };
`;

/**
 * Determines whether the buttons should be hidden.
 *
 * @param {Object} props The component's props.
 * @returns {boolean} True if buttons should be hidden.
 */
const areButtonsHidden = function( props ) {
	return props.marksButtonStatus === "hidden";
};

/**
 * Returns an AnalysisResult component.
 *
 * @param {object} props Component props.
 *
 * @returns {ReactElement} The rendered AnalysisResult component.
 */
export const AnalysisResult = ( props ) => {
	return (
		<AnalysisResultBase>
			<ScoreIcon
				icon={ props.icon }
				color={ props.bulletColor }
				size="13px"
			/>
			<AnalysisResultText suppressedText={ props.suppressedText }>
				{ props.hasBetaBadgeLabel && <BetaBadge /> }
				<span dangerouslySetInnerHTML={ { __html: props.text } } />
			</AnalysisResultText>
			{
				props.hasMarksButton && ! areButtonsHidden( props ) &&
				<IconButtonToggle
					marksButtonStatus={ props.marksButtonStatus }
					className={ props.marksButtonClassName }
					onClick={ props.onButtonClickMarks }
					id={ props.buttonIdMarks }
					icon="eye"
					pressed={ props.pressed }
					ariaLabel={ props.ariaLabelMarks }
				/>
			}
			{
				props.hasEditButton && props.isPremium &&
				<IconCTAEditButton
					className={ props.editButtonClassName }
					onClick={ props.onButtonClickEdit }
					id={ props.buttonIdEdit }
					icon="edit"
					ariaLabel={ props.ariaLabelEdit }
				/>
			}
		</AnalysisResultBase>
	);
};

AnalysisResult.propTypes = {
	icon: PropTypes.string,
	text: PropTypes.string.isRequired,
	suppressedText: PropTypes.bool,
	bulletColor: PropTypes.string.isRequired,
	hasMarksButton: PropTypes.bool.isRequired,
	hasEditButton: PropTypes.bool,
	buttonIdMarks: PropTypes.string.isRequired,
	buttonIdEdit: PropTypes.string,
	pressed: PropTypes.bool.isRequired,
	ariaLabelMarks: PropTypes.string.isRequired,
	ariaLabelEdit: PropTypes.string,
	onButtonClickMarks: PropTypes.func.isRequired,
	onButtonClickEdit: PropTypes.func,
	marksButtonStatus: PropTypes.string,
	marksButtonClassName: PropTypes.string,
	editButtonClassName: PropTypes.string,
	hasBetaBadgeLabel: PropTypes.bool,
	isPremium: PropTypes.bool,
};

AnalysisResult.defaultProps = {
	icon: "circle",
	suppressedText: false,
	marksButtonStatus: "enabled",
	marksButtonClassName: "",
	editButtonClassName: "",
	hasBetaBadgeLabel: false,
	hasEditButton: false,
	buttonIdEdit: "",
	ariaLabelEdit: "",
	onButtonClickEdit: noop,
	isPremium: false,
};

export default AnalysisResult;
