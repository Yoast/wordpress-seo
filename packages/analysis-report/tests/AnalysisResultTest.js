import React from "react";
import renderer from "react-test-renderer";
import { noop } from "lodash";
import AnalysisResult from "../src/AnalysisResult.js";

test( "the AnalysisResult component matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "You're doing great!" }
			score="good"
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with html in the text matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "You're doing <b>great!</b>" }
			score="good"
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with disabled buttons matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "You're doing great!" }
			score="good"
			marksButtonStatus={ "disabled" }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with hidden buttons matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "You're doing great!" }
			score="good"
			marksButtonStatus={ "hidden" }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with suppressed text matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			suppressedText={ true }
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "You're doing great!" }
			score="good"
			marksButtonStatus={ "hidden" }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with a beta badge label matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "This is beta!" }
			score="good"
			marksButtonStatus={ "hidden" }
			hasBetaBadgeLabel={ true }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the AnalysisResult component with a activated premium matches the snapshot", () => {
	const component = renderer.create(
		<AnalysisResult
			ariaLabelMarks="SEOResult"
			ariaLabelEdit="SEOResultEdit"
			bulletColor="blue"
			buttonIdMarks="Result button"
			buttonIdEdit="Result button edit"
			pressed={ true }
			hasMarksButton={ true }
			hasEditButton={ true }
			onButtonClickMarks={ noop }
			onButtonClickEdit={ noop }
			text={ "This is beta!" }
			score="good"
			marksButtonStatus={ "hidden" }
			hasBetaBadgeLabel={ true }
			isPremium={ true }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
