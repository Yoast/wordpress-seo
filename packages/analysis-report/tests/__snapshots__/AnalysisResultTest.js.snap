// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AnalysisResult component matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c5 {
  width: 18px;
  height: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c4 {
  box-sizing: border-box;
  min-width: 32px;
  display: inline-block;
  border: 1px solid #ccc;
  background-color: #a4286a;
  box-shadow: inset 0 2px 0 rgba( 93,35,122,0.7 );
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: 23px;
}

.c4:hover {
  border-color: #fff;
}

.c4:disabled {
  background-color: #f7f7f7;
  box-shadow: none;
  border: none;
  cursor: default;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "You're doing great!",
        }
      }
    />
  </p>
  <button
    aria-label="SEOResult"
    aria-pressed={true}
    className="c4"
    disabled={false}
    id="Result button"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="c5 yoast-svg-icon yoast-svg-icon-eye"
      fill="#fff"
      focusable="false"
      role="img"
      size="18px"
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1664 960q-152-236-381-353 61 104 61 225 0 185-131.5 316.5t-316.5 131.5-316.5-131.5-131.5-316.5q0-121 61-225-229 117-381 353 133 205 333.5 326.5t434.5 121.5 434.5-121.5 333.5-326.5zm-720-384q0-20-14-34t-34-14q-125 0-214.5 89.5t-89.5 214.5q0 20 14 34t34 14 34-14 14-34q0-86 61-147t147-61q20 0 34-14t14-34zm848 384q0 34-20 69-140 230-376.5 368.5t-499.5 138.5-499.5-139-376.5-368q-20-35-20-69t20-69q140-229 376.5-368t499.5-139 499.5 139 376.5 368q20 35 20 69z"
      />
    </svg>
  </button>
</li>
`;

exports[`the AnalysisResult component with a activated premium matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c5 {
  width: 18px;
  height: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c4 {
  box-sizing: border-box;
  min-width: 32px;
  display: inline-block;
  border: 1px solid #ccc;
  background-color: #f7f7f7;
  box-shadow: #ccc;
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: "24px";
}

.c4:hover {
  border-color: #fff;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      className="yoast-badge yoast-beta-badge"
    >
      Beta
    </span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "This is beta!",
        }
      }
    />
  </p>
  <button
    aria-label="SEOResultEdit"
    className="c4"
    id="Result button edit"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="c5 yoast-svg-icon yoast-svg-icon-edit"
      fill="#555"
      focusable="false"
      role="img"
      size="18px"
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M491 1536l91-91-235-235-91 91v107h128v128h107zm523-928q0-22-22-22-10 0-17 7l-542 542q-7 7-7 17 0 22 22 22 10 0 17-7l542-542q7-7 7-17zm-54-192l416 416-832 832h-416v-416zm683 96q0 53-37 90l-166 166-416-416 166-165q36-38 90-38 53 0 91 38l235 234q37 39 37 91z"
      />
    </svg>
  </button>
</li>
`;

exports[`the AnalysisResult component with a beta badge label matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      className="yoast-badge yoast-beta-badge"
    >
      Beta
    </span>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "This is beta!",
        }
      }
    />
  </p>
</li>
`;

exports[`the AnalysisResult component with disabled buttons matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c5 {
  width: 18px;
  height: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c4 {
  box-sizing: border-box;
  min-width: 32px;
  display: inline-block;
  border: 1px solid #ccc;
  background-color: #a4286a;
  box-shadow: inset 0 2px 0 rgba( 93,35,122,0.7 );
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: 23px;
}

.c4:hover {
  border-color: #fff;
}

.c4:disabled {
  background-color: #f7f7f7;
  box-shadow: none;
  border: none;
  cursor: default;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "You're doing great!",
        }
      }
    />
  </p>
  <button
    aria-label="SEOResult"
    aria-pressed={true}
    className="c4"
    disabled={true}
    id="Result button"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="c5 yoast-svg-icon yoast-svg-icon-eye"
      fill="#ddd"
      focusable="false"
      role="img"
      size="18px"
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1664 960q-152-236-381-353 61 104 61 225 0 185-131.5 316.5t-316.5 131.5-316.5-131.5-131.5-316.5q0-121 61-225-229 117-381 353 133 205 333.5 326.5t434.5 121.5 434.5-121.5 333.5-326.5zm-720-384q0-20-14-34t-34-14q-125 0-214.5 89.5t-89.5 214.5q0 20 14 34t34 14 34-14 14-34q0-86 61-147t147-61q20 0 34-14t14-34zm848 384q0 34-20 69-140 230-376.5 368.5t-499.5 138.5-499.5-139-376.5-368q-20-35-20-69t20-69q140-229 376.5-368t499.5-139 499.5 139 376.5 368q20 35 20 69z"
      />
    </svg>
  </button>
</li>
`;

exports[`the AnalysisResult component with hidden buttons matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "You're doing great!",
        }
      }
    />
  </p>
</li>
`;

exports[`the AnalysisResult component with html in the text matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c5 {
  width: 18px;
  height: 18px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c4 {
  box-sizing: border-box;
  min-width: 32px;
  display: inline-block;
  border: 1px solid #ccc;
  background-color: #a4286a;
  box-shadow: inset 0 2px 0 rgba( 93,35,122,0.7 );
  border-radius: 3px;
  cursor: pointer;
  padding: 0;
  height: 23px;
}

.c4:hover {
  border-color: #fff;
}

.c4:disabled {
  background-color: #f7f7f7;
  box-shadow: none;
  border: none;
  cursor: default;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: inherit;
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "You're doing <b>great!</b>",
        }
      }
    />
  </p>
  <button
    aria-label="SEOResult"
    aria-pressed={true}
    className="c4"
    disabled={false}
    id="Result button"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="c5 yoast-svg-icon yoast-svg-icon-eye"
      fill="#fff"
      focusable="false"
      role="img"
      size="18px"
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1664 960q-152-236-381-353 61 104 61 225 0 185-131.5 316.5t-316.5 131.5-316.5-131.5-131.5-316.5q0-121 61-225-229 117-381 353 133 205 333.5 326.5t434.5 121.5 434.5-121.5 333.5-326.5zm-720-384q0-20-14-34t-34-14q-125 0-214.5 89.5t-89.5 214.5q0 20 14 34t34 14 34-14 14-34q0-86 61-147t147-61q20 0 34-14t14-34zm848 384q0 34-20 69-140 230-376.5 368.5t-499.5 138.5-499.5-139-376.5-368q-20-35-20-69t20-69q140-229 376.5-368t499.5-139 499.5 139 376.5 368q20 35 20 69z"
      />
    </svg>
  </button>
</li>
`;

exports[`the AnalysisResult component with suppressed text matches the snapshot 1`] = `
.c1 {
  width: 13px;
  height: 13px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 {
  min-height: 24px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  margin: 3px 11px 0 0;
}

.c3 {
  margin: 0 16px 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: rgba(30,30,30,0.5);
}

<li
  className="c0"
>
  <svg
    aria-hidden={true}
    className="c1 yoast-svg-icon yoast-svg-icon-circle c2"
    fill="blue"
    focusable="false"
    role="img"
    size="13px"
    viewBox="0 0 1792 1792"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1664 896q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"
    />
  </svg>
  <p
    className="c3"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "You're doing great!",
        }
      }
    />
  </p>
</li>
`;
